src/js/tracks/text-track-cue-list.js

```diff
@@ -57,14 +106,47 @@ TextTrackCueList.prototype.setCues_ = function(cues) {
       });
     }
   };
+  defineProp.call(this, this.cues_.length-1);
+  this.length_ = this.cues_.length;
+};
+
+TextTrackCueList.prototype.removeCue_ = function(removeCue) {
+  let removed = false;
+
+  for (let i = 0, l = this.cues_.length; i < l; i++) {
+    let cue = this.cues_[i];
+    if (cue === removeCue) {
+      this.cues_.splice(i, 1);
+      removed = true;
+    }
+  }
 
-  if (oldLength < l) {
-    i = oldLength;
+  if (removed) {
+    this.length_ = this.cues_.length;
+    if(this.optimized_){
+      this.optimizeAllCues_();
+    }
+  }
+};
+
+TextTrackCueList.prototype.getActiveCuesByTime = function(time) {
+  let cues = this.quickCues_[time/100|0]||[];
+  let active = [];
 
-    for(; i < l; i++) {
-      defineProp.call(this, i);
+  for (let i = 0, l = cues.length; i < l; i++) {
+    let cue = cues[i];
```

Comment:
I mended `addCues_` and `removeCue_` functions checking for optimization to avoid that. 

`addCue_` was created to avoid optimizing too often when a WebVTT is parsed.
Before this PR, when a cue was added, every cues were pulled from the track and added again. This took a lot of time when there is a huge number of cues and optimization took also a lot longer.

Unless someone mess directly with the cue table it should not have any issue.

I will update my PR with documentation about `getActiveCuesByTime_` tomorrow.